batavia/types/Dict.js

```diff
@@ -4,305 +4,11 @@
 
 batavia.types.Dict = function() {
     function Dict(args, kwargs) {
-        Object.call(this);
-        if (args) {
-            this.update(args);
-        }
+        batavia.types.JSDict.call(this, args, kwargs);
     }
 
-    Dict.prototype = Object.create(Object.prototype);
+    Dict.prototype = Object.create(batavia.types.JSDict.prototype);
```

Comment:
Sure thing.

The JSDict will still be necessary moving forward. It is probably going to be faster than a roll-our-own dictionary for things like our variable dictionaries. Also, we'll want to be able to have a Dict that is backed by a JS object for when we want to interact with other JS functions that aren't part of batavia (or as part of our translation).

Or maybe you were asking why we need our own Dict implementation? That's easy enough -- Javascript object's don't differentiate between two things whose `toString()` is the same, like `"1"` and `1`, or anything else that might be printed out as `1`. Javascript object keys are always strings, but Python allows any immutable type to be a key, and doesn't care about their string representations.

So, we're going to need our own Python-compliant dictionary (and likewise, set and frozenset) at some point. This is moving in that direction.